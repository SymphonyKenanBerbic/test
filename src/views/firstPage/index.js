/*jshint esversion: 6 */
/*jslint node: true */
"use strict";

import React, { Component } from 'react';
import {Button} from "react-bootstrap";

const _defaultValue = "test";
export default class FirstPage extends Component{
    constructor(){
        super();
        this.state = {index: 0, value: _defaultValue};

        this.onClick = this.onClick.bind(this);
    }

    componentDidMount(){
        let {index} = this.state;
        this.setState({value:_defaultValue + index, index: 1});
    }

    async onClick() {
        let {index} = this.state;
        return new Promise((resolve) => this.setState({value: _defaultValue + index, index: ++index, showElement: index === 2}, resolve));
    }

    render(){
        let {value, index, showElement} = this.state;
        return (<div>test, {value}
            {showElement && (<div>Novi Div</div>)}
            <br/>
            <Button onClick={this.onClick} >Test</Button>
        </div>)
    }
}