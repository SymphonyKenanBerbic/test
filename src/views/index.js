/*jshint esversion: 6 */
/*jslint node: true */
"use strict";

import FirstPage from "./firstPage";
import Dashboard from "./dashboard";

export {
    FirstPage,
    Dashboard
};