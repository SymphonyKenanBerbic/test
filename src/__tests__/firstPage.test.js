import React from 'react';
import {FirstPage} from "../views";
import renderer from 'react-test-renderer';


test('renders without crashing', async () => {
    const component = renderer.create(<FirstPage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('test on click action', async () => {
    const component = renderer.create(<FirstPage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(tree.children).toHaveLength(4);
    expect(tree.children[1]).toMatch("test0");

    await component.getInstance().onClick();
    tree = component.toJSON();

    expect(tree.children[1]).toMatch("test1");

    await component.getInstance().onClick();

    tree = component.toJSON();
    expect(tree.children[1]).toMatch("test2");
    expect(tree.children).toHaveLength(4);
    console.log(tree);
    // re-rendering
});