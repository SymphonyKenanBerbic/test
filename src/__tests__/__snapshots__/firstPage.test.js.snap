// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  test, 
  test0
  <br />
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Test
  </button>
</div>
`;

exports[`test on click action 1`] = `
<div>
  test, 
  test0
  <br />
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Test
  </button>
</div>
`;
